bpf/lib/nodeport.h

```diff
@@ -1687,7 +1687,7 @@ int tail_nodeport_nat_ipv4(struct __ctx_buff *ctx)
 		 * BPF calls as we use tail calls) and complexity, hence this is
 		 * done inside a tail call here.
 		 */
-		if (dir == NAT_DIR_INGRESS) {
+		if (dir == NAT_DIR_INGRESS && ret == DROP_NAT_NO_MAPPING) {
```

Comment:
Check the code again - I think this change is changing the existing behavior of the SNAT. After this change, for the dir=ingress we will start dropping packets which previously were ignored due to `NAT_PUNT_TO_STACK`.

Reply:
I wanted to just use  `if ret == DROP_INVALID` to avoid the tail call to ` CILIUM_CALL_IPV4_FROM_LXC` which is early doing a `revalidate_data_pull()`.

Could you let me know what's the meaning of `NAT_PUNT_TO_STACK`?

Reply:
`NAT_PUNT_TO_STACK` means that the packet should not be SNAT-ed/rev-SNAT-ed. Basically, it is a packet which proto the SNAT engine cannot handle.

Coming back to your original changes, have you actually measured whether it reduces the verifier complexity?

Reply:
Not really, I supposed that as it's calling a `tail call` and then `revalidate_data_pull`.

My understanding is that:  `tail call` are limited in number, add verifier complexity to the prog and bring overhead, and even if `revalidate_data_pull` is called on a new `tail call` there are no reasons to make that extra work. Sounds right?

Do you have an easy way to measure verifier complexity? I was trying with 
```
sudo test/bpf/verifier-test.sh -v -f | grep -e "^Prog" -e processed -e "Proccessed" -e "^=>" -e "over limit"
```

But no luck so far.